javaslang/src/main/java/javaslang/collection/JavaConverters.java

```diff
@@ -104,30 +85,32 @@ boolean isMutable() {
             this.mutable = mutable;
         }
 
-        boolean isMutable() {
+        protected boolean isMutable() {
             return mutable;
         }
 
+        // DEV-NOTE: used within package to access the (original) persistent collection
         C getDelegate() {
             return delegate;
         }
 
-        C setDelegate(C newDelegate) {
+        @SuppressWarnings("unchecked")
+        protected C setDelegate(Traversable<?> newDelegate) {
             if (mutable) {
                 final C previousDelegate = delegate;
-                delegate = newDelegate;
+                delegate = (C) newDelegate;
```

Comment:
The types are ok to the point where we call methods on a Javaslang collection of type C:

```java
<T, C extends Seq<T> test(C seq) {
    // we know that the result is of type C but we get a Seq<T>
    seq.map(t -> t);
}
```

I see no other possibility than doing an unsafe cast. My first version had casts all over the View implementation. But we can do that unsafe cast in a central place, which seems to be better.

I agree that this is a fragile contruct. But we deal only with basic j.u.Collection (and j.u. Map) operations like add(), put(), remove(), ... It should be safe.